docs/architecture.rst

```diff
@@ -149,66 +155,85 @@ if any of the environment variables *SNF_IMAGE_DEV*,
+needed files for cloud-init to use the NoCloud datasource. This task will not
+run unless *SNF_IMAGE_PROPERTY_CLOUD_INIT* is set.
+
 **InstallUnattend**: Installs the Unattend.xml files on Windows instances. This
 is needed by Windows in order to perform an unattended setup. The
-*SNF_IMAGE_TARGET* variables needs to be present for this task to run.
+*SNF_IMAGE_TARGET* variables needs to be present for this task to run. This
+task will not run if *SNF_IMAGE_PROPERTY_CLOUD_INIT* is set.
 
 **FilesystemResizeMounted**: For Windows VMs this task injects a script into
 the VM's file system that will enlarge the last file system to cover up the
 whole partition. The script will run during the specialize pass of the Windows
 setup. For Linux VMs this task is used to extend the last file system in case
 its type is Btrfs or XFS, since those file systems require to be mounted in
 order to resize them. If the *SNF_IMAGE_TARGET* variable is missing, the task
-will fail.
+will fail. This task will not run if *SNF_IMAGE_PROPERTY_CLOUD_INIT* is set.
+Cloud-init will perform the file system live resize.
 
 **AddSwap**: Formats the swap partition added by *FixPartitionTable* task and
 adds an appropriate swap entry in the system's ``/etc/fstab``. The script will
 only run if *SNF_IMAGE_PROPERTY_SWAP* is present and will fail if
-*SNF_IMAGE_TARGET* in not defined.
+*SNF_IMAGE_TARGET* in not defined. Under cloud-init this task will enable swap
+using the the *Mounts* module.
 
 **AssignHostname**: Assigns or changes the hostname of the instance. The task
 will fail if the Linux distribution is not supported and ``/etc/hostname`` is
 not present on the file system. For now, we support Debian, Red Hat, Slackware,
 SUSE and Gentoo derived distributions. The hostname is read from
 *SNF_IMAGE_HOSTNAME* variable. In addition to the latter, *SNF_IMAGE_TARGET* is
-also required.
+also required. For cloud-init, the configuration is performed by setting the
+*local-hostname* key of the datasource's metadata.
 
 **ChangeMachineId**: On Linux instances, this script will generate a new random
 machine ID and will place it in ``/etc/machine-id``. For more info check
 `here <https://www.freedesktop.org/software/systemd/man/machine-id.html>`_. The
 task will fail if *SNF_IMAGE_TARGET* is missing.
 
-**ChangePassword**: Changes the password for a list of existing users. On Linux 
-systems this is accomplished by directly altering the instance's
-``/etc/shadow`` file. On Windows systems a script is injected into the VM's
-hard disk. This script will be executed during the specialize pass of the
-Windows setup. On \*BSD systems ``/etc/master.passwd`` is altered,
+**ChangePassword**: Changes the authentication credentials for a list of
+existing users. On Linux systems this is accomplished by directly altering the
+instance's ``/etc/shadow`` file. On Windows systems a script is injected into
+the VM's hard disk. This script will be executed during the specialize pass of
```

Comment:
This is irrelevant to cloud-init and it's the "specialize pass": https://technet.microsoft.com/en-us/library/cc749307(v=ws.10).aspx